src/xrpld/rpc/detail/TransactionSign.cpp

```diff
@@ -467,6 +467,13 @@ transactionPreProcessImpl(
 
         if (!tx_json.isMember(jss::Flags))
             tx_json[jss::Flags] = tfFullyCanonicalSig;
+
+        if (!tx_json.isMember(jss::NetworkID))
+        {
```

Comment:
"Hint":  getCurrentNetworkFee / checkFee  have 4  paramenters -  `config / feeTrack / txQ / app.` First 3 are always derived from the `app`.  You can remove them from parameter list, leave only app and retrieve other from the app inside `checkFee`

Reply:
I'd rather not do any major refactors in this PR.

Reply:
IMO it is small refactoring, but up to you

Reply:
I can do it next week but would prefer it as a separate PR